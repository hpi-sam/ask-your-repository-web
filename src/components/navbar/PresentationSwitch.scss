.PresentationSwitch {
  background-color: transparent;

  &:focus {
    outline: 0;
  }

  &::-moz-focus-inner {border:0;}

  &--active {
    .PresentationSwitch__outer {
      width: 47px;
      background-position: 0 0;
      background-color: #57BB00;
    }

    .PresentationSwitch__inner {
      transform: translateX(23px);
      background-color: #ffffff;
    }
  }

  &__outer {
    vertical-align: middle;
    width: 47px;
    height: 24px;
    border-radius: 999px;
    background-color: #00B0F2;
    transition-duration: .4s;
    transition-property: background-color, box-shadow;
    cursor: pointer;
  }

  &__inner {
    float: left;
    width: 23px; height: 23px;
    border-radius: inherit;
    background: #ffffff;
    transition-timing-function: cubic-bezier(1,0,0,1);
    transition-duration: 0.4s;
    transition-property: transform, background-color;
    pointer-events: none;
    margin-top: 1px;
    margin-left: 1px;
  }
}
