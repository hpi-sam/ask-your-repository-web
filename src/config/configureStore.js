// @flow
import { applyMiddleware, createStore } from 'redux';
import type { Dispatch } from 'redux';
import { persistStore, persistReducer } from 'redux-persist-erksch';
import storage from 'redux-persist-erksch/lib/storage';
import { routerMiddleware } from 'connected-react-router';
import { createBrowserHistory } from 'history';
import { middleware as flashMiddleware } from 'redux-flash';
import thunk from 'redux-thunk';
import socketioMiddleware from '../middleware/socketioMiddleware';
import axiosAuthMiddleware from './axiosAuthMiddleware';
import createRootReducer from '../state/rootReducer';
import type { AppState } from '../state/AppState';
import type { Action } from '../state/Action';

export const history = createBrowserHistory();

const persistConfig = {
  key: 'root',
  storage,
  whitelist: ['activeTeam', 'presentationMode', 'auth'],
};

function configureStore() {
  let middleware = [
    thunk,
    flashMiddleware(),
    routerMiddleware(history),
    socketioMiddleware(),
    axiosAuthMiddleware(),
  ];

  if (process.env.NODE_ENV === 'development') {
    const { logger } = require('redux-logger');
    middleware = [...middleware, logger];
  }

  const rootReducer = createRootReducer(history);
  const persistedReducer: any = persistReducer(persistConfig, rootReducer);

  const store = createStore<AppState, Action, Dispatch<Action>>(
    persistedReducer,
    applyMiddleware(...middleware),
  );

  const persistor = persistStore(store);

  return { store, persistor };
}

export default configureStore;
