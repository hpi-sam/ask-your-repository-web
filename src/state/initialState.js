// @flow

// This file is only for state related testing

import type { AppState } from './AppState';
import { initialState as presentation } from './presentation/presentation.reducer';
import { initialState as teamSidebar } from './team_sidebar/teamSidebar.reducer';
import { initialState as presentationMode } from './presentation_mode/presentationMode.reducer';
import { initialState as activeTeam } from './active_team/activeTeam.reducer';
import { initialState as auth } from './auth/auth.reducer';

const initialState: AppState = {
  flash: { messages: [] },
  presentation,
  activeTeam,
  teamSidebar,
  presentationMode,
  auth,
};

export default initialState;
