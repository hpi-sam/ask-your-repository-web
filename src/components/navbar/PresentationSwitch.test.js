// @flow
import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import initialState from '../../state/initialState';
import PresentationSwitch from './PresentationSwitch';
import { turnOnPresentationMode, turnOffPresentationMode } from '../../state/presentation_mode/presentationMode.actionCreators';

const mockStore = configureStore();

describe('<PresentationSwitch />', () => {
  let wrapper;
  let store;

  beforeEach(() => {
    store = mockStore(() => initialState);
    wrapper = mount((
      <Provider store={store}>
        <PresentationSwitch />
      </Provider>
    )).find('PresentationSwitch');
  });

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('presentation mode is off', () => {
    beforeAll(() => {
      initialState.presentationMode.isActive = false;
    });

    it('turns on presentation mode on click', () => {
      wrapper.find('button').simulate('click');
      expect(store.getActions()).toEqual([turnOnPresentationMode()]);
    });
  });

  describe('presentation mode is on', () => {
    beforeAll(() => {
      initialState.presentationMode.isActive = true;
    });

    it('turns off presentation mode on click', () => {
      wrapper.find('button').simulate('click');
      expect(store.getActions()).toEqual([turnOffPresentationMode()]);
    });
  });
});
