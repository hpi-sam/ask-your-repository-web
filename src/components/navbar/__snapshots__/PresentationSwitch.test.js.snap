// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PresentationSwitch /> renders correctly 1`] = `
<PresentationSwitch
  dispatch={[Function]}
  isPresentationModeOn={false}
>
  <button
    className="PresentationSwitch"
    data-tip="Synchronized Presentation Mode"
    onClick={[Function]}
    type="button"
  >
    <div
      className="PresentationSwitch__outer"
    >
      <div
        className="PresentationSwitch__inner"
      />
    </div>
  </button>
  <ReactTooltip
    effect="solid"
    insecure={true}
    place="bottom"
    resizeHide={true}
    wrapper="div"
  >
    <div
      className="__react_component_tooltip place-bottom type-dark "
      data-id="tooltip"
    />
  </ReactTooltip>
</PresentationSwitch>
`;
